src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,244 @@
+                        connection.prepareStatement(updateCarRequest)) {
+            updateCarStatement.setLong(1, car.getManufacturer().getId());
+            updateCarStatement.setString(2, car.getModel());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update a car " + car, e);
+        }
+        deleteDrivers(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement =
+                        connection.prepareStatement(deleteCarRequest)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long id) {
+        String getCarsForDriverRequest = "SELECT cars.id, cars.manufacturer_id, cars.model "
+                + "FROM cars JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "WHERE cars_drivers.driver_id = ? AND cars.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsForDriverStatement =
+                        connection.prepareStatement(getCarsForDriverRequest)) {
+            getCarsForDriverStatement.setLong(1, id);
+            ResultSet resultSet = getCarsForDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarWithoutManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars for driver with id " + id, e);
+        }
+        cars.forEach(car -> {
+            car.setDrivers(getDriversForCar(car.getId()));
+        });
+        return cars;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+        manufacturer.setName(resultSet.getString("name"));
+        manufacturer.setCountry(resultSet.getString("country"));
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Car car = new Car();
+        car.setId(id);
+        car.setManufacturer(manufacturer);
+        car.setModel(model);
+        return car;
+    }
+
+    // I use this method because I have no manufacturers in result set
+    // when I want to get all cars by driver
```

Comment:
don't add comments in the future for explanation of your code, it's better to tag me in your private chat